custom_model_runner/datarobot_drum/drum/language_predictors/r_predictor/r_predictor.py

```diff
@@ -102,6 +102,59 @@ def _get_sparse_colnames(kwargs):
             sparse_colnames = ro.rinterface.NULL
         return sparse_colnames
 
+    def _replace_sanitized_class_names(self, data):
+        """ Match prediction data labels to project class labels.
+        Note that this contains only logic specific to R name
+        sanitization and relies on marshal_predictions() for
+        language neutral cases like matching floats and bools """
+
+        # get class labels
+        if (
+            self._r_positive_class_label is not ro.rinterface.NULL
+            and self._r_negative_class_label is not ro.rinterface.NULL
+        ):
+            class_labels = [self._r_negative_class_label, self._r_positive_class_label]
+        elif self._r_class_labels is not None:
+            class_labels = self._r_class_labels
+        else:
+            raise DrumCommonException("Class labels not available for classification.")
+
+        prediction_labels = data.columns
+
+        # if the labels match then do nothing
+        if set(prediction_labels) == set(class_labels):
+            return data
+
+        # check for match after make.names is applied to class labels
+        sanitized_class_labels = ro.r["make.names"](class_labels)
+        if set(prediction_labels) == set(sanitized_class_labels):
+            if len(set(sanitized_class_labels)) != len(sanitized_class_labels):
+                raise DrumCommonException("Class label names are ambiguous.")
```

Comment:
nit: pull this out of the if statement.

Reply:
I'm not sure what you mean here.

Reply:
there isn't a reason to nest this inside the above if statement it seems... IDK this is mostly an aesthetic choice, less nested ifs are more readable.

Reply:
if moved, I think it might raise an error incorrectly in the case where prediction labels aren't a sanitized version of the request labels and applying make.names to them will introduce duplicates.